public/app/util/router.js

```diff
@@ -0,0 +1,98 @@
+(function () {
+    'use strict';
+
+    define(
+        [
+            'lodash',
+            'knockout',
+            'page'
+        ],
+        function (_, ko, page) {
+            var runHook, attach, current,
+                applicationContainer = document.getElementById('application-container');
+
+            // Run the given named hook for the given vm.
+            runHook = function (vm, hook, container, callback) {
+                callback = callback || _.noop;
+                if (_.isFunction(vm[hook])) {
+                    return vm[hook](container, callback);
+                }
+                callback();
+            };
+
+            // Attach the view and view model specified by paths, using the given router context.
+            attach = function (viewPath, viewModelPath, context) {
+                var container, vm, _attach;
+
+                _attach = function () {
```

Comment:
I could have used promises here to flatten it, but it's not too bad.  If I decide to bring them in then I'll convert this part, but for now I'm happy with callbacks.